libraries/rush-sdk/src/proxy.ts

```diff
@@ -0,0 +1,288 @@
+  /**
+   * The task has completed successfully. No further events will be fired.
+   */
+  | 'succeeded'
+  /**
+   * The task was aborted by the caller. No further events will be fired.
+   */
+  | 'aborted'
+  /**
+   * The task has failed due to an error. No further events will be fired.
+   */
+  | 'failed';
+
+/**
+ * Event options for {@link ILoadSdkAsyncOptions.onNotifyEvent}
+ * @public
+ */
+export interface ISdkCallbackEvent {
+  /**
+   * Indicates the current state of the operation.
+   */
+  state: SdkCallbackState;
+
+  /**
+   * Allows the caller to display log information about the operation.
+   */
+  logMessage: IProgressBarCallbackLogMessage | undefined;
+
+  /**
+   * Allows the caller to display a progress bar for long-running operations.
+   *
+   * @remarks
+   * If a long-running operation is required, then `progressBarPercent` will
+   * start at 0.0 and count upwards and finish at 100.0 if the operation completes
+   * successfully.  If the long-running operation has not yet started, or
+   * is not required, then the value will be `undefined`.
+   */
+  progressBarPercent: number | undefined;
+}
+
+/**
+ * Type of {@link ILoadSdkAsyncOptions.onNotifyEvent}
+ * @public
+ */
+export type SdkNotifyEventCallback = (sdkEvent: ISdkCallbackEvent) => void;
+
+/**
+ * Options for {@link rushSdkProxy.loadAsync}
+ * @public
+ */
+export interface ILoadSdkAsyncOptions {
+  /**
+   * The folder to start from when searching for the Rush workspace configuration.
+   * If this folder does not contain a `rush.json` file, then each parent folder
+   * will be searched.  If `rush.json` is not found, then the SDK fails to load.
+   */
+  rushJsonSearchFolder?: string;
+
+  /**
+   * A cancellation token that the caller can use to prematurely abort the operation.
+   */
+  abortSignal?: AbortSignal;
+
+  /**
+   * Allows the caller to monitor the progress of the operation.
+   */
+  onNotifyEvent?: SdkNotifyEventCallback;
+}
+
+/**
+ * Exposes operations that control how the `@microsoft/rush-lib` engine is
+ * located and loaded.
+ * @public
+ */
+// eslint-disable-next-line @typescript-eslint/no-namespace
+export namespace rushSdkProxy {
+  function _checkForCancel(
+    abortSignal: AbortSignal,
+    onNotifyEvent: SdkNotifyEventCallback | undefined,
+    progressBarPercent: number | undefined
+  ): boolean {
+    if (!abortSignal.aborted) {
+      return true;
+    }
+
+    if (onNotifyEvent) {
+      onNotifyEvent({
+        state: 'aborted',
+        logMessage: {
+          messageType: 'info',
+          message: `The operation was cancelled`
+        },
+        progressBarPercent
+      });
+    }
+
+    return false;
+  }
+
+  /**
+   * Manually load the Rush engine based on rush.json found for `rushJsonSearchFolder`.
+   * This API supports an optional cancellation token and callback for status updates
+   * during the operation.
+   */
+  export async function loadAsync(options?: ILoadSdkAsyncOptions): Promise<void> {
```

Comment:
```suggestion
  export async function loadAsync(options?: ILoadSdkAsyncOptions): Promise<typeof RushLibModule> {
```

Reply:
I think ultimately it would be a better design for the entire Rush API to be returned as an object rather than an import, but that is a major redesign.  Introducing a better way would be more compelling if we at the same time removed the old way.

Also there is a technical hitch, that `typeof RushLibModule` would require `proxy.d.ts` to import `rush-sdk.d.ts` which is going to cause trouble